client/api/request.js

```diff
@@ -1,13 +1,19 @@
 import axios from 'axios';
 
-// TODO: read this from configuration.
-const BASE_URL = '/api/v1/';
-
-// Instance of axios to be used for all API requests.
-const client = axios.create({
-  baseURL: BASE_URL,
+const config = {
+  baseURL: process.env.API_PATH,
   withCredentials: true,
   headers: { 'Content-Type': 'application/json' }
+};
+
+// Instance of axios to be used for all API requests.
+const client = axios.create(config);
+
+Object.defineProperty(client, 'base', {
+  get() {
```

Comment:
Why using this approach to get singleton ?

Reply:
Why it is lazily created or? 😕